.github/workflows/build.yml

```diff
@@ -15,23 +15,33 @@ jobs:
     - name: Prepare
       run: |
         sudo apt-get update
-        sudo apt-get install -y libssl-dev libcurl4-openssl-dev uncrustify libyaml-dev
+        sudo apt-get install -y libssl-dev libcurl4-openssl-dev uncrustify
```

Comment:
Can we find a solution to install `libyaml-dev` by `apt-get` instead of building from source code ?
If this part changes, https://github.com/kubernetes-client/c/blob/master/README.md should also be changed, but it will  be inconvenient for user.

Reply:
It's not possible because apt's version does not come with CMake configuration scripts required for find_package() to discover the libraries and includes. More recent versions of the code the library do include CMake configuration scripts in addition to pkg-config, and that's why they are discoverable.

Reply:
So we have multiple options here:
1- Use pkg-config for finding the package, something I don't recommend.
2- Change the README.md to include libyaml building instructions.
3- Add conditional detection that can work with both pkg-config and CMake configuration scripts.

Note that the approach used before which uses no find_package() works only on Unix based systems. The correct way to do this with CMake is using find_package() and configuration scripts.

Reply:
What's the version of apt/Ubuntu to meet the requirement of CMake/find_package() ?

Hi @brendandburns 
What's your opinion ?

Reply:
I'm using Ubuntu 20.04 and the available libyaml-dev version is 0.2.2.  I'm not sure which apt version has the CMake configuration files. I just know for sure that apt's version only install pkg-config configuration files and no CMake scripts.

Reply:
The reason that CMake cannot `find_package(libyaml)` is the dpkg package `libyaml-dev` in Ubuntu/apt now does not have the file `libyaml-dev-config.cmake` and CMake has not an internal cmake file for `libyaml-dev`.

We only can wait or ask developer of `libyaml-dev` to package `*.config.cmake` in the future.

Reply:
There is another option in fact: we can use the pkg-config files that ship with apt’s version of Libyans-dev. Do you want me to do that?

Reply:
If we need clone source code, please only download a stable version, e.g.
```
git clone https://libwebsockets.org/repo/libwebsockets --depth 1 --branch v4.2-stable
```

Reply:
Hi @ahmedyarub 

I don't want to ship config.cmake for libyaml-dev in this repo/package. It should be the responsibility of libyaml-dev.

Let's involve @brendandburns to discuss the solution.

Reply:
The solution that only works on Unix based systems is when we do not use find_package(). In that case make or the compiler will just look into standard library paths. If I use the pkg-config scripts that come with CMake it should work everywhere. However, I still prefer using CMake configuration scripts (the ones generated when we build libyaml manually). They seem to be more stable when used across platforms and especially package managers like vcpkg.